sklearn/decomposition/_kernel_pca.py

```diff
@@ -439,7 +439,9 @@ def fit(self, X, y=None):
         self.gamma_ = 1 / X.shape[1] if self.gamma is None else self.gamma
         self._centerer = KernelCenterer().set_output(transform="default")
         K = self._get_kernel(X)
-        self._fit_transform(K)
+        # safe to perform in place operations on K because a copy was made before if
```

Comment:
So I think that was on the thing I was not sure about: `K` is a new value right, or there is a way somehow that `K` is a view on `X` for a particular kernel and that transforming `K` in place will change `X`?

Reply:
K is X in the case of a precomputed kernel and `copy=False`, The goal of `copy=False` is to allow to change X in place

Reply:
OK, the `'precomputed'` case was not obvious to me, can you add this to the comment, and also find a way to mention that `copy_X` has already made a copy if needed (this is something you probably mentioned in the issue but that wasn't obvious to me either)

Reply:
Indeed let's add a comment to make that case explicit.